### DISCUSSION TITLE
Accessing Data from Mongo DB Nested Document within Data Provider

### DISCUSSION THREAD

**Discussion by JayP718** [Budibase/budibase#9087] [Help] [answered]
Hi all,

Say I have a Mongo DB document like the following.

{
   "department":"marketing",
   "manager":"jane",
   "items":[
      "desk",
      "chair"
   ],
   "child_departments":[
      {
         "department":"hr",
         "manager":"mary",
         "items":[
            "desk",
            "chair"
         ]
      }
   ]
}

And I want to access the elements from the parent department and the child departments at once in the data provider. I know I can use the transform within the custom queries but I'm wondering if there's a way to specify in the data provider level to reduce database pulls and still be able to show parent child data.

Thank you!

**Comment by melohagan** [comment, marked as answer]
Hey @JayP718 

If you used aggregations that would do the filtering and transformations server side: https://docs.budibase.com/docs/mongodb#aggregations
